src/lib/SaferNames/Syntax.hs

```diff
@@ -154,135 +157,62 @@ type Con n = PrimCon (Atom n)
 type Op  n = PrimOp  (Atom n)
 type Hof n = PrimHof (Atom n)
 
+type IndexStructure = Nest Binder
+
+type AtomSubstVal = SubstVal TypedBinderInfo Atom :: E -> E
+
 -- === top-level modules ===
 
 type SourceName = T.Text
 
 data SourceNameMap n = SourceNameMap
-  { fromSourceNameMap :: M.Map SourceName (Name n)}
+  { fromSourceNameMap :: M.Map SourceName (Name UnitE n)}
 
 data Module n where
   Module :: IRVariant
          -> Nest Decl n l
-         -> BindingsFrag l l'
+         -> ScopeFrag l l'
          -> SourceNameMap l'
          -> Module n
```

Comment:
I know this was checked in before, but can you please document the fields here? I don't remember what they mean anymore